### PR TITLE
fix(core): Plan templated pipelines before triggering from start tasks

### PR DESCRIPTION
**Pull request by louisjimenez**
While fixing this [issue](https://github.com/spinnaker/spinnaker/issues/4623), I noticed that the pipeline run stage fails when a pipeline triggers a dependent templated pipeline. This was due to the plan step being omitted by the StartPipelineTask. These changes plan for v2 pipelines and allows them to trigger properly.

### CODE REVIEW

File: orca-front50/src/test/groovy/com/netflix/spinnaker/orca/front50/tasks/StartPipelineTaskSpec.groovy [hunk 1]
```diff
@@ -16,9 +16,11 @@
```
Comment (louisjimenez) [inline orca-front50/src/test/groovy/com/netflix/spinnaker/orca/front50/tasks/StartPipelineTaskSpec.groovy]: @ezimanyi here are the changes for this test

Comment (ezimanyi) [reply orca-front50/src/test/groovy/com/netflix/spinnaker/orca/front50/tasks/StartPipelineTaskSpec.groovy]: Looks good! :+1:

**Comment by louisjimenez** [comment]
@spinnakerbot cherry-pick 1.15

**Comment by louisjimenez** [comment]
@spinnakerbot cherry-pick 1.14

**Pull request merged**